conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureUrlDecodingHandler.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import io.undertow.UndertowOptions;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.PathTemplateMatch;
+import io.undertow.util.URLUtils;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Backport of the Undertow URLDecodingHandler proposed for from Undertow 2.0.18.Final.
+ * This may be replaced with the Undertow URLDecodingHandler once
+ * <a href="https://github.com/undertow-io/undertow/pull/710">Undertow pull/710</a> merges.
+ * Fixes matched path parameter decoding which should not allow form encoding.
+ */
+final class ConjureUrlDecodingHandler implements HttpHandler {
```

Comment:
It feels a tiny bit funky to be backporting unmerged PRs - how urgent do you think the form encoding bugfix actually is - would it be tolerable to just stay on vanilla undertow for a bit longer until that PR merges?